inc/saisr.h

```diff
@@ -0,0 +1,490 @@
+    SAI_SR_BSID_ATTR_CUSTOM_RANGE_END
+
+} sai_sr_bsid_attr_t;
+
+/**
+ * @brief Enum defining Transit or Source types
+ */
+typedef enum _sai_sr_sidlist_type_t
+{
+    /** Insertion of Segment Route Policy */
+    SAI_SR_SIDLIST_TYPE_INSERT,
+
+    /** Encapsulation in a Segment Route Policy and Source */
+    SAI_SR_SIDLIST_TYPE_ENCAPS,
+
+    /** Encapsulation in a Segment Route Policy and Ethernet */
+    SAI_SR_SIDLIST_TYPE_ENCAPS_L2,
+
+    /** Custom range base value */
+    SAI_SR_SIDLIST_TYPE_CUSTOM_RANGE_BASE = 0x10000000
+
+} sai_sr_sidlist_type_t;
+
+/**
+ * @brief Attribute data for Segment Route Segment ID List Entry
+ */
+typedef enum _sai_sr_sidlist_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_SR_SIDLIST_ATTR_START = 0x00000000,
+
+    /**
+     * @brief Transit or Source Type
+     *
+     * @type sai_sr_sidlist_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_SR_SIDLIST_ATTR_TYPE = SAI_SR_SIDLIST_ATTR_START,
+
+    /**
+     * @brief Binding Segment ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_SR_BSID
+     */
+    SAI_SR_SIDLIST_ATTR_SR_BSID,
+
+    /**
+     * @brief Member weights
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 1
+     */
+    SAI_SR_SIDLIST_ATTR_WEIGHT,
+
+    /**
+     * @brief Number of Segments supported on device for source
+     *
+     * @type sai_uint8_t
+     * @flags READ_ONLY
+     */
+    SAI_SR_SIDLIST_ATTR_SUPPORTED_NUM_SEGMENTS,
+
+    /**
+     * @brief List of TLV types supported on device for source
+     *
+     * @type sai_s32_list_t sai_tlv_type_t
+     * @flags READ_ONLY
+     */
+    SAI_SR_SIDLIST_ATTR_SUPPORTED_TLV_TYPE,
+
+    /**
+     * @brief List of TLVs for Source
+     *
+     * @type sai_tlv_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_SR_SIDLIST_ATTR_TLV_LIST,
+
+    /**
+     * @brief List of Segments for Source / Transit
+     *
+     * @type sai_segment_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_SR_SIDLIST_ATTR_SEGMENT_LIST,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_SR_SIDLIST_ATTR_END,
+
+    /** Custom range base value */
+    SAI_SR_SIDLIST_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_SR_SIDLIST_ATTR_CUSTOM_RANGE_END
+} sai_sr_sidlist_attr_t;
+
+/**
+ * @brief Policy / Endpoint entry
+ */
+typedef struct _sai_sr_pe_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief IPv6 Destination
+     */
+    sai_ip_address_t destination;
+
+    /**
+     * @brief Color to match on policy
+     */
+    sai_uint32_t policy_color;
```

Comment:
in this PR you changed this struct 2 times already, it seems to me that this design needs a little bit more polishing before final version ? also policy color should that be an attribute ? does this needs to really be in the struct? we should keep struct members to minimum